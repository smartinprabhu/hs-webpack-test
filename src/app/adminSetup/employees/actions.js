import { CALL_API } from '../../../middleware/api';

export const GET_EMPLOYEES_INFO = 'GET_EMPLOYEES_INFO';
export const GET_EMPLOYEES_INFO_SUCCESS = 'GET_EMPLOYEES_INFO_SUCCESS';
export const GET_EMPLOYEES_INFO_FAILURE = 'GET_EMPLOYEES_INFO_FAILURE';

export const GET_EMPLOYEES_COUNT = 'GET_EMPLOYEES_COUNT';
export const GET_EMPLOYEES_COUNT_SUCCESS = 'GET_EMPLOYEES_COUNT_SUCCESS';
export const GET_EMPLOYEES_COUNT_FAILURE = 'GET_EMPLOYEES_COUNT_FAILURE';

export const GET_CATEGORIES_GROUP_INFO = 'GET_CATEGORIES_GROUP_INFO';
export const GET_CATEGORIES_GROUP_INFO_SUCCESS = 'GET_CATEGORIES_GROUP_INFO_SUCCESS';
export const GET_CATEGORIES_GROUP_INFO_FAILURE = 'GET_CATEGORIES_GROUP_INFO_FAILURE';

export const GET_EMPLOYEE_DETAILS = 'GET_EMPLOYEE_DETAILS';
export const GET_EMPLOYEE_DETAILS_SUCCESS = 'GET_EMPLOYEE_DETAILS_SUCCESS';
export const GET_EMPLOYEE_DETAILS_FAILURE = 'GET_EMPLOYEE_DETAILS_FAILURE';

export const GET_NEIGHBOUR_DETAILS = 'GET_NEIGHBOUR_DETAILS';
export const GET_NEIGHBOUR_DETAILS_SUCCESS = 'GET_NEIGHBOUR_DETAILS_SUCCESS';
export const GET_NEIGHBOUR_DETAILS_FAILURE = 'GET_NEIGHBOUR_DETAILS_FAILURE';

export const LOAD_NEIGHBOUR_DETAILS = 'LOAD_NEIGHBOUR_DETAILS';
export const LOAD_NEIGHBOUR_DETAILS_SUCCESS = 'LOAD_NEIGHBOUR_DETAILS_SUCCESS';
export const LOAD_NEIGHBOUR_DETAILS_FAILURE = 'LOAD_NEIGHBOUR_DETAILS_FAILURE';

export const GET_UPDATE_EMPLOYEE_INFO = 'GET_UPDATE_EMPLOYEE_INFO';
export const GET_UPDATE_EMPLOYEE_INFO_SUCCESS = 'GET_UPDATE_EMPLOYEE_INFO_SUCCESS';
export const GET_UPDATE_EMPLOYEE_INFO_FAILURE = 'GET_UPDATE_EMPLOYEE_INFO_FAILURE';

export function getEmployeesInfo(company, model, limit, offset, states, customFilters, sortByValue, sortFieldValue) {
  let payload = `domain=[["company_id","in",[${company}]]`;
  if (states && states.length > 0) {
    payload = `${payload},["registration_status","in",${JSON.stringify(states)}]`;
  }
  if (customFilters && customFilters.length > 0) {
    payload = `${payload},${customFilters}`;
  }
  payload = `${payload}]&model=${model}&fields=["work_email","name","work_phone","registration_status","employee_id_seq","department_id"]&limit=${limit}&offset=${offset}`;

  if (sortFieldValue && sortFieldValue.length > 0) {
    payload = `${payload}&order=${sortFieldValue}`;
  }
  if (sortByValue && sortByValue.length > 0) {
    payload = `${payload} ${sortByValue}`;
  }

  return {
    [CALL_API]: {
      endpoint: `search_read?${payload}`,
      types: [GET_EMPLOYEES_INFO, GET_EMPLOYEES_INFO_SUCCESS, GET_EMPLOYEES_INFO_FAILURE],
      method: 'GET',
      payload,
    },
  };
}

export function getEmployeesCount(company, model, states, customFilters) {
  let payload = `domain=[["company_id","in",[${company}]]`;
  if (states && states.length > 0) {
    payload = `${payload},["registration_status","in",${JSON.stringify(states)}]`;
  }
  if (customFilters && customFilters.length > 0) {
    payload = `${payload},${customFilters}`;
  }
  payload = `${payload}]&model=${model}&fields=["__count"]&groupby=[]`;
  return {
    [CALL_API]: {
      endpoint: `read_group?${payload}`,
      types: [GET_EMPLOYEES_COUNT, GET_EMPLOYEES_COUNT_SUCCESS, GET_EMPLOYEES_COUNT_FAILURE],
      method: 'GET',
      payload,
    },
  };
}

export function getCategoryGroupsInfo(company, model) {
  const payload = `domain=[["company_id","in",[${company}]]]&model=${model}&fields=["category_id"]&groupby=["category_id"]`;
  return {
    [CALL_API]: {
      endpoint: `read_group?${payload}`,
      types: [GET_CATEGORIES_GROUP_INFO, GET_CATEGORIES_GROUP_INFO_SUCCESS, GET_CATEGORIES_GROUP_INFO_FAILURE],
      method: 'GET',
      payload,
    },
  };
}

export function getEmployeeDetails(id, modelName) {
  return {
    [CALL_API]: {
      endpoint: `read/${modelName}?ids=[${id}]&fields=[]`,
      types: [GET_EMPLOYEE_DETAILS, GET_EMPLOYEE_DETAILS_SUCCESS, GET_EMPLOYEE_DETAILS_FAILURE],
      method: 'GET',
      id,
    },
  };
}

export function getNeighbourhoodsInfo(values, modelName) {
  let payload = `domain=[["id","in",[${values}]]]`;
  payload = `${payload}&model=${modelName}&fields=["path_name","space_name"]`;
  payload = `${payload}&limit=50&offset=0&order=sort_sequence ASC`;
  return {
    [CALL_API]: {
      endpoint: `search_read?${payload}`,
      types: [GET_NEIGHBOUR_DETAILS, GET_NEIGHBOUR_DETAILS_SUCCESS, GET_NEIGHBOUR_DETAILS_FAILURE],
      method: 'GET',
      payload,
    },
  };
}

export function loadNeighbourhoodsInfo(company, model, sortBy, sortField) {
  let payload = `domain=[["company_id","in",[${company}]],["asset_category_id","ilike","Neighbourhood"]`;
  payload = `${payload}]&model=${model}&fields=["path_name","space_name"]&limit=50&offset=0&order=id ASC`;

  if (sortField && sortField.length > 0) {
    payload = `${payload}&order=${sortField}`;
  }
  if (sortBy && sortBy.length > 0) {
    payload = `${payload} ${sortBy}`;
  }
  return {
    [CALL_API]: {
      endpoint: `search_read?${payload}`,
      types: [LOAD_NEIGHBOUR_DETAILS, LOAD_NEIGHBOUR_DETAILS_SUCCESS, LOAD_NEIGHBOUR_DETAILS_FAILURE],
      method: 'GET',
      payload,
    },
  };
}

export function updateEmployeeDataInfo(id, result, modelName) {
  const payload = {
    ids: `[${id}]`, values: result,
  };
  return {
    [CALL_API]: {
      endpoint: `write/${modelName}`,
      types: [GET_UPDATE_EMPLOYEE_INFO, GET_UPDATE_EMPLOYEE_INFO_SUCCESS, GET_UPDATE_EMPLOYEE_INFO_FAILURE],
      method: 'PUT',
      payload,
    },
  };
}
