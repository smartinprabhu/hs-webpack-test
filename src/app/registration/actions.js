import { CALL_API } from '../../middleware/api';

export const CHECK_EMAIL = 'CHECK_EMAIL';
export const CHECK_EMAIL_SUCCESS = 'CHECK_EMAIL_SUCCESS';
export const CHECK_EMAIL_FAILURE = 'CHECK_EMAIL_FAILURE';
export const REGISTER_USER = 'REGISTER_USER';
export const REGISTER_USER_SUCCESS = 'REGISTER_USER_SUCCESS';
export const REGISTER_USER_FAILURE = 'REGISTER_USER_FAILURE';
export const RESET_REGISTRATION = 'RESET_REGISTRATION';
export const SHOW_REGISTRATION = 'SHOW_REGISTRATION';

export const getMailValidation = (mailId, token, tenantId) => ({
  [CALL_API]: {
    endpoint: 'employee/validate_email',
    types: [CHECK_EMAIL, CHECK_EMAIL_SUCCESS, CHECK_EMAIL_FAILURE],
    method: 'POST',
    payload: {
      tenant_id: tenantId,
      email_id: mailId,
      token,
    },
  },
});

export const validateOtp = (userData) => ({
  [CALL_API]: {
    endpoint: `employee/validate_otp?session_token=${userData.session}&otp=${userData.otp}&email_id=${userData.email}&token=${userData.token}`,
    types: [CHECK_EMAIL, CHECK_EMAIL_SUCCESS, CHECK_EMAIL_FAILURE],
    method: 'POST',
  },
});

export const userRegistration = (userData) => ({
  [CALL_API]: {
    endpoint: `employee/register?token=${userData.token}
    &session_token=${userData.session}&first_name=${userData.firstName}
    &last_name=${userData.lastName}&work_email=${userData.email}&work_phone=${userData.workNumber}
    &password=${userData.password}&retype_password=${userData.repeatPassword}`,
    types: [REGISTER_USER, REGISTER_USER_SUCCESS, REGISTER_USER_FAILURE],
    method: 'POST',
  },
});

export function resetRegistrationProcess() {
  return {
    type: RESET_REGISTRATION,
  };
}

export function showRegistrationForm() {
  return {
    type: SHOW_REGISTRATION,
  };
}
