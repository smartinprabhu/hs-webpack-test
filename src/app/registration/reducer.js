import {
  CHECK_EMAIL, CHECK_EMAIL_SUCCESS, CHECK_EMAIL_FAILURE,
  REGISTER_USER, REGISTER_USER_SUCCESS, REGISTER_USER_FAILURE, RESET_REGISTRATION, SHOW_REGISTRATION,
} from './actions';

const initialState = {
  mailResponse: {},
  userRegistartionResponse: {},
  errorMessage: {},
  registerLoading: false,
  sessionToken: '',
};

function reducer(state, action) {
  if (typeof state === 'undefined') {
    return initialState;
  }
  switch (action.type) {
    case CHECK_EMAIL:
      return {
        ...state,
        registerLoading: true,
      };

    case CHECK_EMAIL_SUCCESS:
      return {
        ...state,
        registerLoading: false,
        mailResponse: action.payload.data,
        errorMessage: {},
        sessionToken: action.payload.data.session_token,
      };

    case CHECK_EMAIL_FAILURE:
      return {
        ...state,
        registerLoading: false,
        errorMessage: action.error.data,
      };

    case REGISTER_USER:
      return {
        ...state,
        registerLoading: true,
        errorMessage: {},
      };

    case REGISTER_USER_SUCCESS:
      return {
        ...state,
        registerLoading: false,
        userRegistartionResponse: action.payload.data,
        errorMessage: {},
      };

    case REGISTER_USER_FAILURE:
      return {
        ...state,
        registerLoading: false,
        errorMessage: action.error.data,
        userRegistartionResponse: {},
      };

    case RESET_REGISTRATION:
      return {
        ...state,
        registerLoading: false,
        mailResponse: {},
        errorMessage: {},
      };

    case SHOW_REGISTRATION:
      return {
        ...state,
        registerLoading: false,
        userRegistartionResponse: {},
        errorMessage: {},
      };
    default: return {
      ...state,
    };
  }
}

export default reducer;
