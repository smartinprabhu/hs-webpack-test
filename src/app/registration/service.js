import {
  getMailValidation, validateOtp, userRegistration, resetRegistrationProcess, showRegistrationForm,
} from './actions';

export function getMailStatus(userData, token, tenantId) {
  return (dispatch) => {
    dispatch(getMailValidation(userData.email, token, tenantId));
  };
}

export function getOtpValidation(userData) {
  return (dispatch) => {
    dispatch(validateOtp(userData));
  };
}

export function registerUser(userData) {
  return (dispatch) => {
    dispatch(userRegistration(userData));
  };
}

export function resetRegistration() {
  return (dispatch) => {
    dispatch(resetRegistrationProcess());
  };
}

export function showRegistration() {
  return (dispatch) => {
    dispatch(showRegistrationForm());
  };
}
