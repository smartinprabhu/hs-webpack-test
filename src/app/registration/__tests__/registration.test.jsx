import React from 'react';
import configureMockStore from 'redux-mock-store';
import { render, fireEvent, cleanup } from '@testing-library/react';
// import { Provider, useDispatch, useSelector } from 'react-redux';
import * as redux from 'react-redux';
import { act } from '@testing-library/react-hooks';
import fetchMock from 'fetch-mock';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/dom';

import MailRegistration from '../mailRegistration';
import UserRegistration from '../registartion';
import * as services from '../service';
import * as actions from '../actions';
import reducer from '../reducer';
import api from '../../../middleware/api';
import store from '../../../store';

afterEach(() => {
  fetchMock.restore();
  cleanup();
});

function renderWithRedux(component) {
  const { Provider } = redux;
  return {
    ...render(<Provider store={store}>{component}</Provider>),
  };
}

describe('registration snapshots', () => {
  it('mail registration renders without crashing', () => {
    const { asFragment } = renderWithRedux(<MailRegistration />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('registration snapshots', () => {
  it('mail registration renders without crashing', () => {
    const registerObject = {
      sessionToken: 'abc',
      mail: 'user@example.com',
    };
    const { asFragment } = renderWithRedux(<UserRegistration registerObject={registerObject} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('registration actions, reducers, middleware', () => {
  it('it should mock getMailValidation function', () => {
    actions.getMailValidation = jest.fn();
    const getMailValidation = (data,data2,data3) => actions.getMailValidation(data,data2,data3);
    getMailValidation('user@example.com', 'token-local', 'tenantId');
    expect(actions.getMailValidation).toBeCalled();
  });

  it('it should mock resetRegistrationProcess function', () => {
    actions.resetRegistrationProcess = jest.fn();
    const resetRegistrationProcess = (data,data2,data3) => actions.resetRegistrationProcess(data,data2,data3);
    resetRegistrationProcess('user@example.com', 'token-local', 'tenantId');
    expect(actions.resetRegistrationProcess).toBeCalled();
  });

  const create = () => {
    const storeCreate = jest.fn();
    const next = jest.fn();
    const invoke = (action) => api(storeCreate)(next)(action);
    return { next, invoke };
  };

  it('passes through CHECK_EMAIL action', () => {
    const { next, invoke } = create();
    const action = { type: actions.CHECK_EMAIL };
    invoke(action);
    expect(next).toHaveBeenCalledWith(action);
  });

  it('passes through CHECK_EMAIL_SUCCESS action', () => {
    const { next, invoke } = create();
    const action = { type: actions.CHECK_EMAIL_SUCCESS };
    invoke(action);
    expect(next).toHaveBeenCalledWith(action);
  });

  it('passes through CHECK_EMAIL_FAILURE action', () => {
    const { next, invoke } = create();
    const action = { type: actions.CHECK_EMAIL_FAILURE };
    invoke(action);
    expect(next).toHaveBeenCalledWith(action);
  });

  it('passes through RESET_REGISTRATION action', () => {
    const { next, invoke } = create();
    const action = { type: actions.RESET_REGISTRATION };
    invoke(action);
    expect(next).toHaveBeenCalledWith(action);
  });

  const initialState = {
    mailResponse: {},
    userRegistartionResponse: {},
    errorMessage: {},
    registerLoading: false,
  };

  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState);
  });

  it('should handle CHECK_EMAIL', () => {
    expect(reducer(initialState, {
      type: 'CHECK_EMAIL',
    })).toEqual({
      ...initialState,
      registerLoading: true,
    });
  });

  it('should handle CHECK_EMAIL_SUCCESS', () => {
    expect(reducer(initialState, {
      type: 'CHECK_EMAIL_SUCCESS',
      payload: { data: '' },
    })).toEqual({
      ...initialState,
      registerLoading: false,
      mailResponse: '',
      errorMessage: {},
    });
  });

  it('should handle CHECK_EMAIL_FAILURE', () => {
    expect(reducer(initialState, {
      type: 'CHECK_EMAIL_FAILURE',
      error: { data: '' },
    })).toEqual({
      ...initialState,
      registerLoading: false,
      errorMessage: '',
    });
  });

  it('should handle REGISTER_USER', () => {
    expect(reducer(initialState, {
      type: 'REGISTER_USER',
    })).toEqual({
      ...initialState,
      registerLoading: true,
    });
  });

  it('should handle REGISTER_USER_SUCCESS', () => {
    expect(reducer(initialState, {
      type: 'REGISTER_USER_SUCCESS',
      payload: { data: '' },
    })).toEqual({
      ...initialState,
      registerLoading: false,
      userRegistartionResponse: '',
      errorMessage: {},
    });
  });

  it('should handle REGISTER_USER_FAILURE', () => {
    expect(reducer(initialState, {
      type: 'REGISTER_USER_FAILURE',
      error: { data: '' },
    })).toEqual({
      ...initialState,
      registerLoading: false,
      errorMessage: '',
      userRegistartionResponse: {},

    });
  });

  it('should handle RESET_REGISTRATION', () => {
    expect(reducer(initialState, {
      type: 'RESET_REGISTRATION',
    })).toEqual({
      ...initialState,
      registerLoading: false,
      mailResponse: {},
      errorMessage: {},
    });
  });

  it('should handle SHOW_REGISTRATION', () => {
    expect(reducer(initialState, {
      type: 'SHOW_REGISTRATION',
    })).toEqual({
      ...initialState,
      registerLoading: false,
      userRegistartionResponse: {},
      errorMessage: {},
    });
  });

  it('should check dispatch is called', () => {
    const useDispatchSpy = jest.spyOn(redux, 'useDispatch');
    const mockDispatchFn = jest.fn();
    useDispatchSpy.mockReturnValue(mockDispatchFn);

    mockDispatchFn(services.resetRegistration());
    expect(mockDispatchFn).toBeCalled();
  });

  it('should check all service', () => {
    const resetRegistrationSpy = jest.spyOn(services, 'resetRegistration');
    const registerUserSpy = jest.spyOn(services, 'registerUser');
    const getOtpValidationSpy = jest.spyOn(services, 'getOtpValidation');
    const getMailStatusSpy = jest.spyOn(services, 'getMailStatus');

    services.resetRegistration();
    expect(resetRegistrationSpy).toBeCalled();

    services.registerUser();
    expect(registerUserSpy).toBeCalled();

    services.getOtpValidation();
    expect(getOtpValidationSpy).toBeCalled();

    services.getMailStatus('user@example.com, token, tenantId');
    expect(getMailStatusSpy).toBeCalled();
  });

  it('should check mail registration field', () => {
    act(() => {
      renderWithRedux(<MailRegistration />);
    });

    const email = screen.getByLabelText('Email', { selector: 'input' });
    act(() => {
      fireEvent.change(email, {
        target: {
          value: 'user@example.com',
        },
      });
    });

    expect(email).toHaveAttribute('id', 'email');
    expect(email).toHaveAttribute('value', 'user@example.com');
  });

  it('should check otp field', () => {
    const initialState1 = {
      mailRegister: {
        mailResponse: { message: 'OTP has been sent successfully.' },
        userRegistartionResponse: {},
        errorMessage: {},
        loginLoading: false,
        registerLoading: false,
      },
    };

    const onSubmit = jest.fn();
    const mockStore = configureMockStore();
    const { Provider } = redux;
    const { container } = render(<Provider store={mockStore(initialState1)}><MailRegistration onSubmit={onSubmit} /></Provider>);

    const otpField = screen.getByLabelText('OTP', { selector: 'input' });
    act(() => {
      fireEvent.change('otp', {
        target: {
          value: '1234',
        },
      });
    });

    expect(otpField).toHaveAttribute('id', 'otp');
    expect(otpField).toHaveAttribute('value', '1234');

    const backButton = screen.getByText('BACK');
    act(() => {
      fireEvent.click(backButton);
    });
    expect(services.resetRegistration).toHaveBeenCalled();

    act(() => {
      fireEvent.change('email', {
        target: {
          value: 'user@example.com',
        },
      });

      fireEvent.change('otp', {
        target: {
          value: '1234',
        },
      });
      const regButton = screen.getByText('REGISTER');
      fireEvent.click(regButton);
    });
    expect(container).toBeInTheDocument();

    act(() => {
      fireEvent.change('email', {
        target: {
          value: 'user@example.com',
        },
      });

      fireEvent.change('otp', {
        target: {
          value: '',
        },
      });
      const regButton1 = screen.getByText('REGISTER');
      fireEvent.click(regButton1);
    });
    expect(container).toBeInTheDocument();
  });

  it('should check all mail validation conditions', () => {
    const initialState2 = {
      mailRegister: {
        mailResponse: { message: '' },
        userRegistartionResponse: {},
        errorMessage: {},
        loginLoading: false,
        registerLoading: false,
      },
    };

    const onSubmit = jest.fn();
    const mockStore = configureMockStore();
    const { Provider } = redux;
    const { container } = render(<Provider store={mockStore(initialState2)}><MailRegistration onSubmit={onSubmit} /></Provider>);

    act(() => {
      fireEvent.change('email', {
        target: {
          value: 'user@example.com',
        },
      });

      const registerButton = screen.getByText('REGISTER');
      fireEvent.click(registerButton);
    });
    expect(container).toBeInTheDocument();

    act(() => {
      fireEvent.change('email', {
        target: {
          value: 'user.com',
        },
      });

      const regButton = screen.getByText('REGISTER');
      fireEvent.click(regButton);
    });
    expect(container).toBeInTheDocument();
  });

  it('should check otp validation', () => {
    const initialState3 = {
      mailRegister: {
        mailResponse: { message: 'OTP has been sent successfully.' },
        userRegistartionResponse: {},
        errorMessage: {},
        loginLoading: false,
        registerLoading: false,
      },
    };

    const onSubmit = jest.fn();
    const mockStore = configureMockStore();
    const { Provider } = redux;
    const { container } = render(<Provider store={mockStore(initialState3)}><MailRegistration onSubmit={onSubmit} /></Provider>);

    const otpField = screen.getByLabelText('OTP', { selector: 'input' });
    const email = screen.getByLabelText('Email', { selector: 'input' });
    act(() => {
      fireEvent.change(email, {
        target: {
          value: 'user@example.com',
        },
      });

      fireEvent.change(otpField, {
        target: {
          value: '123',
        },
      });

      const registerButton = screen.getByText('REGISTER');
      fireEvent.click(registerButton);
    });
    expect(container).toBeInTheDocument();
  });

  it('should check all dom testing', () => {
    const initialState4 = {
      mailRegister: {
        mailResponse: { message: 'OTP sent succesfully!.' },
        registerLoading: true,
      },
    };

    const onSubmit = jest.fn();
    const { Provider } = redux;
    const mockStore = configureMockStore();
    const { container } = render(<Provider store={mockStore(initialState4)}><MailRegistration onSubmit={onSubmit} /></Provider>);
    expect(container).toBeInTheDocument();
  });

  it('should check all dom testing1', () => {
    const initialState5 = {
      mailRegister: {
        mailResponse: { message: 'OTP sent succesfully!.' },
        registerLoading: false,
        errorMessage: {
          data: { message: 'Invalid OTP' },
        },
      },
    };

    const onSubmit = jest.fn();
    const { Provider } = redux;
    const mockStore = configureMockStore();
    const { container } = render(<Provider store={mockStore(initialState5)}><MailRegistration onSubmit={onSubmit} /></Provider>);

    // const text = screen.getByText(/invalid OTP/i, { exact: true });
    // expect(text).toHaveTextContent('invalid OTP');
    expect(container).toBeInTheDocument();
  });

  it('should check all dom testing2', () => {
    const initialState6 = {
      mailRegister: {
        mailResponse: { message: 'OTP sent succesfully!.' },
        registerLoading: false,
        errorMessage: {
          data: { message: 'Invalid OTP' },
          error: { message: 'Invalid Email' },
        },
      },
    };

    const onSubmit = jest.fn();
    const { Provider } = redux;
    const mockStore = configureMockStore();
    const { container } = render(<Provider store={mockStore(initialState6)}><MailRegistration onSubmit={onSubmit} /></Provider>);

    expect(container).toBeInTheDocument();
  });

  it('should check onSubmit', () => {
    const initialState7 = {
      mailRegister: {
        mailResponse: { message: 'OTP Confirmed Successfully.' },
        userRegistartionResponse: {},
        errorMessage: {},
        loginLoading: false,
        registerLoading: false,
      },
    };

    const registerObject = {
      sessionToken: 'abc',
      mail: 'user@example.com',
    };

    const mockStore = configureMockStore();
    const { Provider } = redux;
    const onSubmit = jest.fn();
    const { container } = render(<Provider store={mockStore(initialState7)}><UserRegistration registerObject={registerObject} onSubmit={onSubmit} /></Provider>);

    const fname = screen.getByLabelText('First Name', { selector: 'input' });
    const lname = screen.getByLabelText('Last Name', { selector: 'input' });
    const workNum = screen.getByLabelText('Work Number', { selector: 'input' });
    const paswrd = screen.getByLabelText('Password', { selector: 'input' });
    const repeatpaswrd = screen.getByLabelText('Re-type Password', { selector: 'input' });

    act(() => {
      fireEvent.change(fname, {
        target: {
          value: 'fname',
        },
      });

      fireEvent.change(lname, {
        target: {
          value: 'lname',
        },
      });

      fireEvent.change(workNum, {
        target: {
          value: '1234567890',
        },
      });

      fireEvent.change(paswrd, {
        target: {
          value: 'passWord@10',
        },
      });

      fireEvent.change(repeatpaswrd, {
        target: {
          value: 'passWord@10',
        },
      });

      const registerButton = screen.getByText('REGISTER');
      fireEvent.click(registerButton);
    });
    expect(container).toBeInTheDocument();
  });

  it('should check all user registration fields', () => {
    const registerObject = {
      sessionToken: 'abc',
      mail: 'user@example.com',
    };
    const { container, getByText } = renderWithRedux(<UserRegistration registerObject={registerObject} />);
    const fname = screen.getByLabelText('First Name', { selector: 'input' });
    const lname = screen.getByLabelText('Last Name', { selector: 'input' });
    const workNum = screen.getByLabelText('Work Number', { selector: 'input' });
    const paswrd = screen.getByLabelText('Password', { selector: 'input' });
    const repeatpaswrd = screen.getByLabelText('Re-type Password', { selector: 'input' });

    act(() => {
      fireEvent.change(fname, {
        target: {
          value: 'fname',
        },
      });
    });

    expect(fname).toHaveAttribute('id', 'firstName');
    expect(fname).toHaveAttribute('value', 'fname');
    expect(lname).toHaveAttribute('id', 'lastName');
    expect(lname).toHaveAttribute('value', '');
    expect(workNum).toHaveAttribute('id', 'workNumber');
    expect(workNum).toHaveAttribute('value', '');
    expect(paswrd).toHaveAttribute('id', 'password');
    expect(paswrd).toHaveAttribute('value', '');
    expect(repeatpaswrd).toHaveAttribute('id', 'repeatPassword');
    expect(repeatpaswrd).toHaveAttribute('value', '');

    act(() => {
      fireEvent.click(getByText('REGISTER'));
    });
    expect(container).toBeInTheDocument();
  });

  it('should handle actions', () => {
    const configuredMockStore = configureMockStore();
    const mockStore = configuredMockStore({});

    const CHECK_EMAIL = () => ({
      type: 'CHECK_EMAIL',
    });

    mockStore.dispatch(CHECK_EMAIL());
    expect(mockStore.getActions()).toMatchSnapshot();

    const CHECK_EMAIL_SUCCESS = (action) => ({
      type: 'CHECK_EMAIL_SUCCESS',
      action,
    });
    mockStore.dispatch(CHECK_EMAIL_SUCCESS(
      {
        data: {
          data: '',
        },
        error: {},
      },
    ));
    expect(mockStore.getActions()).toMatchSnapshot();

    const CHECK_EMAIL_FAILURE = (action) => ({
      type: 'CHECK_EMAIL_FAILURE',
      action,
    });
    mockStore.dispatch(CHECK_EMAIL_FAILURE(
      {
        data: {},
        error: {
          data: '',
        },
      },
    ));
    expect(mockStore.getActions()).toMatchSnapshot();
  });

  it('should handle SHOW_REGISTRATION action', () => {
    const expectedAction = {
      type: actions.SHOW_REGISTRATION,
    };
    expect(actions.showRegistrationForm()).toEqual(expectedAction);
  });

  it('should handle RESET_REGISTRATION action', () => {
    const expectedAction = {
      type: actions.RESET_REGISTRATION,
    };
    expect(actions.resetRegistrationProcess()).toEqual(expectedAction);
  });
});
