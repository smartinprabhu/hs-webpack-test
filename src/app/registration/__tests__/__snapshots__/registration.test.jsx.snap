// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration actions, reducers, middleware should handle actions 1`] = `
Array [
  Object {
    "type": "CHECK_EMAIL",
  },
]
`;

exports[`registration actions, reducers, middleware should handle actions 2`] = `
Array [
  Object {
    "type": "CHECK_EMAIL",
  },
  Object {
    "action": Object {
      "data": Object {
        "data": "",
      },
      "error": Object {},
    },
    "type": "CHECK_EMAIL_SUCCESS",
  },
]
`;

exports[`registration actions, reducers, middleware should handle actions 3`] = `
Array [
  Object {
    "type": "CHECK_EMAIL",
  },
  Object {
    "action": Object {
      "data": Object {
        "data": "",
      },
      "error": Object {},
    },
    "type": "CHECK_EMAIL_SUCCESS",
  },
  Object {
    "action": Object {
      "data": Object {},
      "error": Object {
        "data": "",
      },
    },
    "type": "CHECK_EMAIL_FAILURE",
  },
]
`;

exports[`registration snapshots mail registration renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <div
      class="login"
    >
      <div
        class="m-0 row"
      >
        <div
          class="bg-white p-4 mt-5 opacity-9 col-sm-4 order-sm-1 offset-sm-7 col-md-4 order-md-1 offset-md-7 col-lg-4 order-lg-1 offset-lg-7"
        >
          <div
            class="m-0 row"
          >
            <div
              class="pl-0 col-sm-8 col-md-8 col-lg-8"
            >
              <h3>
                Register Email
              </h3>
              <span
                class="light-text"
              >
                Please register your email with us.
              </span>
            </div>
            <div
              class="col-sm-4 col-md-4 col-lg-4"
            >
              <img
                alt="Helixsense Portal"
                class="float-right"
                src="test-file-stub"
                width="80"
              />
            </div>
          </div>
          <form
            class="mt-3"
            novalidate=""
          >
            <div
              class="form-group"
            >
              <label
                class="mb-1"
                for="email"
              >
                Email
              </label>
              <input
                class="border-radius-50px form-control"
                id="email"
                name="email"
                placeholder="yourmail@example.com"
                type="email"
                value=""
              />
            </div>
            <div
              class="mb-1"
            >
              <div
                class="row"
              >
                <div
                  class="col-sm-12"
                >
                  <button
                    class="roundCorners p-login float-right ml-2 btn btn-secondary"
                    type="submit"
                  >
                     REGISTER
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`registration snapshots mail registration renders without crashing 2`] = `
<DocumentFragment>
  <div
    class="login"
  >
    <div
      class="m-0 row"
    >
      <div
        class="bg-white p-4 mt-5 opacity-9 col-sm-4 order-sm-1 offset-sm-7 col-md-4 order-md-1 offset-md-7 col-lg-4 order-lg-1 offset-lg-7"
      >
        <div
          class="m-0 row"
        >
          <div
            class="pl-0 col-sm-8 col-md-8 col-lg-8"
          >
            <h3>
              Registration
            </h3>
            <span
              class="light-text"
            >
              Please register yourself with us.
            </span>
          </div>
          <div
            class="col-sm-4 col-md-4 col-lg-4"
          >
            <img
              alt="Helixsense Portal"
              class="float-right"
              src="test-file-stub"
              width="80"
            />
          </div>
        </div>
        <form
          class="mt-3"
        >
          <div
            class="form-group"
          >
            <label
              class="mb-1"
              for="firstName"
            >
              First Name
            </label>
            <input
              class="border-radius-50px form-control"
              id="firstName"
              name="firstName"
              placeholder="First Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="mb-1"
              for="lastName"
            >
              Last Name
            </label>
            <input
              class="border-radius-50px form-control"
              id="lastName"
              name="lastName"
              placeholder="Last Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="mb-1"
              for="workNumber"
            >
              Work Number
            </label>
            <input
              class="border-radius-50px form-control"
              id="workNumber"
              name="workNumber"
              placeholder="Work Number"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="mb-1"
              for="password"
            >
              Password
            </label>
            <input
              class="border-radius-50px form-control"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="mb-1"
              for="repeatPassword"
            >
              Re-type Password
            </label>
            <input
              class="border-radius-50px form-control"
              id="repeatPassword"
              name="repeatPassword"
              placeholder="Re-type Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="mb-1 row"
          >
            <div
              class="col-sm-11 order-sm-11 offset-sm-1"
            >
              <button
                class="roundCorners p-login float-right ml-2 btn btn-secondary"
                type="submit"
              >
                 REGISTER
              </button>
              <button
                class="roundCorners p-login float-right btn btn-secondary"
                type="button"
              >
                BACK
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
